app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java

```diff
@@ -254,9 +254,11 @@ private static boolean updateDislike(AtomicReference<Object> textRef, boolean is
             }
             replacementSpannable = newSpannableWithDislikes(oldSpannable, voteData);
         } else {
-            String leftSegmentedSeparatorString = ReVancedUtils.isRightToLeftTextLayout() ? "\u200F|  " : "|  ";
+            final boolean useCompactLayout = SettingsEnum.RYD_USE_COMPACT_LAYOUT.getBoolean();
+            // if compact layout, use a "half space" character
+            String middleSegmentedSeparatorString = useCompactLayout ? "\u2009 • \u2009" : "  •  ";
 
-            if (oldLikesString.contains(leftSegmentedSeparatorString)) {
+            if (oldLikesString.contains(middleSegmentedSeparatorString)) {
```

Comment:
Currently, this works, but it's a "hacky" way. You can depend on the `video-information` patch and check if the video has changed and determine this way if it is necessary to update the string. Not a necessary change and unrelated to this PR but something to consider in the future.

Reply:
It must check the actual layout string, because during layout the method is called multiple times on the same span.  And layout happens anytime the span first appears on screen (such as scrolling off screen, and then scrolling back up).

So it must check the actual span each time, and not just once per video load.